crates/transpiler/src/equivalence.rs

```diff
@@ -637,14 +635,13 @@ impl EquivalenceLibrary {
     /// will return only the circuits provided.
     pub fn set_entry(
         &mut self,
-        py: Python,
         gate: &PackedOperation,
         params: &[Param],
-        entry: Vec<CircuitFromPython>,
+        entry: Vec<CircuitData>,
     ) -> PyResult<()> {
         for equiv in entry.iter() {
             raise_if_shape_mismatch(gate, equiv)?;
-            raise_if_param_mismatch(py, params, equiv.0.unsorted_parameters(py)?)?;
+            raise_if_param_mismatch(params, equiv.parameters())?;
```

Comment:
Hmm, I think this was an oversight in #14799 this is actually going to sort the parameters in the circuit which isn't really necessary for a mismatch check. What we really need is a Rust method on `CircuitData` that returns the iterator from calling `CircuitData.param_table::iter_symbols()`. It should be a 3 line method:

```rust
pub fn iter_parameters(&self) -> impl Iterator<Item = &Symbol> {
    self.param_table.iter_symbols()
}
```

Reply:
Added in 10cf87a (??)